src/controllers/controllerhidreporttabsmanager.cpp

```diff
@@ -0,0 +1,498 @@
+            QVariant customData = pTable->item(row, 0)->data(Qt::UserRole + 1);
+            if (customData.isValid()) {
+                auto pControl =
+                        reinterpret_cast<hid::reportDescriptor::Control*>(
+                                customData.value<void*>());
+                // Set the control value in the reportData
+                bool success = hid::reportDescriptor::applyLogicalValue(
+                        reportData, *pControl, item->text().toLongLong());
+                if (!success) {
+                    qWarning() << "Failed to set control value for row" << row;
+                    continue;
+                }
+            }
+        }
+    }
+
+    // Send the reportData
+    if (reportType == hid::reportDescriptor::HidReportType::Feature) {
+        pJsProxy->sendFeatureReport(reportId, reportData);
+    } else if (reportType == hid::reportDescriptor::HidReportType::Output) {
+        pJsProxy->sendOutputReport(reportId, reportData);
+    }
+}
+
+void ControllerHidReportTabsManager::populateHidReportTable(
+        QTableWidget* pTable,
+        const hid::reportDescriptor::Report& pReport,
+        hid::reportDescriptor::HidReportType reportType) {
+    // Temporarily disable updates to speed up populating
+    pTable->setUpdatesEnabled(false);
+
+    // Reserve rows up-front
+    const auto& controls = pReport.getControls();
+    pTable->setRowCount(static_cast<int>(controls.size()));
+
+    // Set the delegate once if needed
+    if (reportType != hid::reportDescriptor::HidReportType::Input) {
+        pTable->setItemDelegateForColumn(5, new ValueItemDelegate(pTable));
+    }
+
+    bool showVolatileColumn = (reportType == hid::reportDescriptor::HidReportType::Feature ||
+            reportType == hid::reportDescriptor::HidReportType::Output);
+
+    // Set headers
+    QStringList headers = {QStringLiteral("Byte Position"),
+            QStringLiteral("Bit Position"),
+            QStringLiteral("Bit Size"),
+            QStringLiteral("Logical Min"),
+            QStringLiteral("Logical Max"),
+            QStringLiteral("Value"),
+            QStringLiteral("Physical Min"),
+            QStringLiteral("Physical Max"),
+            QStringLiteral("Unit Scaling"),
+            QStringLiteral("Unit"),
+            QStringLiteral("Abs/Rel"),
+            QStringLiteral("Wrap"),
+            QStringLiteral("Linear"),
+            QStringLiteral("Preferred"),
+            QStringLiteral("Null")};
+    if (showVolatileColumn) {
+        headers << QStringLiteral("Volatile");
+    }
+    headers << QStringLiteral("Usage Page") << QStringLiteral("Usage");
+
+    pTable->setColumnCount(headers.size());
+    pTable->setHorizontalHeaderLabels(headers);
+    pTable->verticalHeader()->setVisible(false);
+
+    // Helpers
+    auto createReadOnlyItem = [](const QString& text, bool rightAlign = false) {
+        auto* item = new QTableWidgetItem(text);
```

Comment:
You may pass here the pTable and create the item with make_parented()

Reply:
There is no overload of QTableWidgetItem, which allows to pass a parent object.